File: core/ImageSmoother/ImageSmoother.java
package com.dbc;

public class ImageSmoother {
    public int[][] imageSmoother(int[][] img) {
        int[][] res = new int[img.length][img[0].length];
        for (int row = 0; row < img.length; row++) {
            for (int col = 0; col < img[0].length; col++) {
                int total = 0, count = 0;
                for (int i = row - 1; i < row + 2; i++) {
                    for (int j = col - 1; j < col + 2; j++) {
                        if (i >= 0 && i < img.length && j >=0 && j < img[0].length) {
                            total += img[i][j];
                            count++;
                        }
                    }
                }
                res[row][col] = total / count;
            }
        }
        return res;
    }
}

File: core/ImageSmoother/ImageSmoother.py
from typing import List


def imageSmoother(img: List[List[int]]) -> List[List[int]]:
    m, n = len(img), len(img[0])
    res = []
    for row in range(0, m):
        res.append([])
        for col in range(0, n):
            total, count = 0, 0
            for i in range(row - 1, row + 2):
                for j in range(col - 1, col + 2):
                    if 0 <= i < m and 0 <= j < n:
                        total += img[i][j]
                        count += 1
            res[row].append(total // count)

    return res
